File: java-receiver-acceptance-tests/src/test/java/com/wittsfamily/rcode/javareceiver_acceptancetests/RCodeIdentifyCommand.java
package com.wittsfamily.rcode.javareceiver_acceptancetests;

import com.wittsfamily.rcode.javareceiver.RCodeLockSet;
import com.wittsfamily.rcode.javareceiver.RCodeOutStream;
import com.wittsfamily.rcode.javareceiver.RCodeResponseStatus;
import com.wittsfamily.rcode.javareceiver.commands.RCodeCommand;
import com.wittsfamily.rcode.javareceiver.parsing.RCodeCommandSequence;
import com.wittsfamily.rcode.javareceiver.parsing.RCodeCommandSlot;

public class RCodeIdentifyCommand implements RCodeCommand {

    @Override
    public void execute(RCodeCommandSlot slot, RCodeCommandSequence sequence, RCodeOutStream out) {
        out.writeField('V', (byte) 0);
        out.continueField((byte) 0);
        out.continueField((byte) 1);
        out.writeBigStringField("Java/AcceptanceTests");
        out.writeStatus(RCodeResponseStatus.OK);
        slot.setComplete(true);
    }

    @Override
    public void finish(RCodeCommandSlot slot, RCodeOutStream out) {
    }

    @Override
    public void setLocks(RCodeLockSet locks) {
    }

    @Override
    public byte getCode() {
        return 0;
    }

    @Override
    public boolean matchesCode(byte[] code) {
        return code.length == 1 && code[0] == getCode();
    }

    @Override
    public int getCodeLength() {
        return 1;
    }

    @Override
    public byte[] getFullCode() {
        return new byte[] { getCode() };
    }

}

File: arm-no-os/src/test/c++/commands/RCodeIdentifyCommand.cpp
/*
 * RCodeEchoCommand.cpp
 *
 *  Created on: 9 Sep 2020
 *      Author: robert
 */

#include "RCodeIdentifyCommand.hpp"

void RCodeIdentifyCommand::execute(RCodeCommandSlot<RCodeParameters> *slot, RCodeCommandSequence<RCodeParameters> *sequence,
        RCodeOutStream<RCodeParameters> *out) {
    out->writeStatus(OK);
    out->writeField('V', (uint8_t) 0);
    out->continueField((uint8_t) 0);
    out->continueField((uint8_t) 1);
    out->writeBigStringField("rcode/arm/no-os/initial-demo");
    slot->setComplete(true);
}
